gpumon/gpumon_interface.ml

```diff
@@ -0,0 +1,28 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+let service_name = "gpu"
+let queue_name = Xcp_service.common_prefix ^ service_name
+let json_path = "/var/xapi/" ^ service_name ^ ".json"
+let xml_path = "/var/xapi/" ^ service_name
+
+type debug_info = string
+
+type pgpu_address = string
+type pgpu_metadata = string
+type dom_id = int
+type vgpu_compatibility = unit (* temporarily unit, until real type is determined *)
+
+external get_pgpu_metadata: debug_info -> pgpu_address -> pgpu_metadata = ""
+external get_pgpu_vm_compatibility: debug_info -> pgpu_metadata -> dom_id -> vgpu_compatibility list = ""
```

Comment:
I have extended the bindings in a way that requires some changes here. To get the pGPU handle, and then the metadata, you will need the `pci.bus_id` in place of `pgpu_address`. Compatibility here can be any type useful to xapi, we can reshape it inside `gpumon` when calling the idl